diff -NaurpB binutils/binutils/stabs.c b/binutils/stabs.c
--- binutils/binutils/stabs.c	2011-03-25 18:46:07.000000000 +0100
+++ b/binutils/stabs.c	2012-08-20 19:17:25.843756270 +0200
@@ -2593,7 +2593,7 @@ parse_stab_members (void *dhandle, struc
 	  bfd_boolean constp, volatilep, staticp;
 	  bfd_vma voffset;
 	  debug_type context;
-	  const char *physname;
+	  const char *physname = NULL;
 	  bfd_boolean varargs;
 
 	  if (look_ahead_type != DEBUG_TYPE_NULL)
