diff -NaurpB glibc-2.17/ports/sysdeps/arm/nptl/pthread_spin_lock.c glibc-2.17.fix/ports/sysdeps/arm/nptl/pthread_spin_lock.c
--- glibc-2.17/ports/sysdeps/arm/nptl/pthread_spin_lock.c	2012-12-25 04:02:13.000000000 +0100
+++ glibc-2.17.fix/ports/sysdeps/arm/nptl/pthread_spin_lock.c	2013-04-28 17:38:05.376436012 +0200
@@ -19,5 +19,5 @@
 
 /* We can't use the normal "#include <nptl/pthread_spin_lock.c>" because
    it will resolve to this very file.  Using "sysdeps/.." as reference to the
-   top level directory does the job.  */
-#include <sysdeps/../nptl/pthread_spin_lock.c>
+   top level directory does the job.  BULLSHIT */
+#include "../../../nptl/pthread_spin_lock.c"
